src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Designer.targets

```diff
@@ -38,6 +38,9 @@ Copyright (C) 2016 Xamarin. All rights reserved.
     <ExtraJarLocation Include="@(_LibraryJars)">
       <Source>LibraryImport</Source>
     </ExtraJarLocation>
+    <ExtraJarLocation Include="@(AndroidJavaLibrary)">
+      <Source>AndroidJavaLibrary</Source>
+    </ExtraJarLocation>
```

Comment:
Is there a NuGet package or something we could add to this test, that would show this problem?

https://github.com/xamarin/xamarin-android/blob/676b131ba2eed1c654fbcb5aee74ce7e0ce0469e/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/DesignerTests.cs#L140

Reply:
I _just_ released this fix in the Xamarin.AndroidX.Migration package -preview03 on the https://nugetized.blob.core.windows.net/androidx/index.json feed.
The next preview it is going to be moved into the actual AndroidX package. So you could wait for that as well, but it will take some time to build: https://github.com/xamarin/AndroidSupportComponents/pull/185

Reply:
To do a test, create a new Android project, remove the support libraries and replace them with AndroidX and watch the designer fail. This is not the migration bits, just plain AndroidX.

Here is a sample (save as zip): ![AndroidXDesigner](https://user-images.githubusercontent.com/1096616/62217161-6ca3d800-b3aa-11e9-869a-4bcd6400aacd.png)

Reply:
Is there a simpler way to see the bug? Could we just add an `@(AndroidJavaLibrary)` to the project, using one of these or something:

https://github.com/xamarin/xamarin-android/blob/676b131ba2eed1c654fbcb5aee74ce7e0ce0469e/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs#L3428-L3442

Then check the library comes out the other end? It looks like the test currently doesn't check the output, other than check it's not blank:

https://github.com/xamarin/xamarin-android/blob/676b131ba2eed1c654fbcb5aee74ce7e0ce0469e/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/DesignerTests.cs#L164

Reply:
@jonathanpeppers that might be it. You just need to add a view or a theme. The bug appears because the Theme is pulling in a whole bunch of bits. Let me see if I can get a better test case.

Reply:
Yup. Just did a test. If you use the `@(AndroidJavaLibrary)` in the project file, it is not added to the output of the designer tasks. Even if you actually add the .jar into the project folder with an include in the .csproj. I believe that it _should_ appear?

Reply:
Yep, for a unit-test purpose just having a random `@(AndroidJavaLibrary)` item to check for should be plenty enough